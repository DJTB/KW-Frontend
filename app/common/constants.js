/* eslint-disable prefer-destructuring */
// https://github.com/mrsteele/dotenv-webpack/issues/70
export const VERSION = process.env.VERSION;
window.KW_VERSION = VERSION;
export const RAVEN_DSN = process.env.RAVEN_DSN;
const DEV_DOMAIN = process.env.DEV_DOMAIN;
const PROD_DOMAIN = 'https://www.kaniwani.com';
export const IS_DEV_ENV = process.env.NODE_ENV === 'development';
export const IS_PROD_ENV = process.env.NODE_ENV === 'production';
const BASE_URL = IS_DEV_ENV ? DEV_DOMAIN : PROD_DOMAIN;
/* eslint-enable prefer-destructuring */

export const API_BASE = `${BASE_URL}/api/v1/`;

export const DATE_FORMAT = 'MMM Do YYYY';
export const TIME_FORMAT = 'hh:mma';
export const DATE_TIME_FORMAT = `${DATE_FORMAT}[,] ${TIME_FORMAT}`;

export const MAX_NOTES_LENGTH = 500;
export const MAX_CONTACT_LENGTH = 1000;

export const WK_SRS_RANKS = {
  ONE: 'APPRENTICE',
  TWO: 'GURU',
  THREE: 'MASTER',
  FOUR: 'ENLIGHTENED',
  FIVE: 'BURNED',
};

export const SRS_RANKS = {
  ZERO: 'UNTRAINED',
  ...WK_SRS_RANKS,
};

export const SRS_RANGES = {
  ZERO: [0],
  ONE: [1, 2, 3, 4],
  TWO: [5, 6],
  THREE: [7],
  FOUR: [8],
  FIVE: [9],
};

export const KEYCODES = {
  SPACE: 32,
  ENTER: 13,
};

export const TAGS = {
  common: 'common',
  MA: 'Martial Arts',
  X: 'Rude/X-Rated',
  abbr: 'Abbreviation',
  adj: 'Adjective',
  'adj-i': 'I Adjective',
  'adj-ix': 'I Adjective',
  'adj-na': 'Na Adjective',
  'adj-no': 'No Adjective',
  'adj-pn': 'Pre-Noun Adjectival',
  'adj-t': 'Taru Adjective',
  'adj-f': 'Prenominal Noun/Verb',
  adv: 'Adverb',
  'adv-to': 'To Adverb',
  arch: 'Archaism',
  ateji: 'Ateji',
  aux: 'Auxiliary',
  'aux-v': 'Auxiliary Verb',
  'aux-adj': 'Auxiliary Adjective',
  Buddh: 'Buddhist',
  chem: 'Chemistry',
  chn: 'Childish',
  col: 'Colloquial',
  comp: 'Computer',
  conj: 'Conjunction',
  'cop-da': 'Copula',
  ctr: 'Counter',
  derog: 'Derogatory',
  eK: 'Kanji Only',
  ek: 'Kana Only',
  exp: 'Expression',
  fam: 'Familiar',
  fem: 'Feminine',
  food: 'Food',
  geom: 'Geometry',
  gikun: 'Gikun/Jukujikun',
  hon: 'Honorific (sonkeigo)',
  hum: 'Humble (kenjougo)',
  iK: 'Irregular Kanji',
  id: 'Idiom',
  ik: 'Irregular Kana',
  int: 'Interjection',
  io: 'Irregular Okurigana',
  iv: 'Irregular Verb',
  ling: 'Linguistics',
  'm-sl': 'Manga Slang',
  male: 'Masculine',
  'male-sl': 'Male Slang',
  math: 'Mathematics',
  mil: 'Military',
  n: 'Noun',
  'n-adv': 'Adverbial Noun',
  'n-suf': 'Noun Suffix',
  'n-pref': 'Noun Prefix',
  'n-t': 'Temporal Noun',
  num: 'Numeric',
  oK: 'Out-dated Kanji',
  obs: 'Obsolete',
  obsc: 'Obscure',
  ok: 'Out-Dated Kana',
  oik: 'Old/Irregular Kana',
  'on-mim': 'Onomatopoeic/Mimetic',
  pn: 'Pronoun',
  poet: 'Poetry',
  pol: 'Polite (teineigo)',
  pref: 'Prefix',
  proverb: 'Proverb',
  prt: 'Particle',
  physics: 'Physics',
  rare: 'Rare',
  sens: 'Sensitive',
  sl: 'Slang',
  suf: 'Suffix',
  uK: 'Usually Kanji',
  uk: 'Usually Kana',
  unc: 'Unclassified',
  yoji: 'Yojijukugo',
  v1: 'Ichidan Verb',
  'v1-s': 'Ichidan Verb',
  'v2a-s': 'Nidan Verb',
  v4h: 'Yodan Verb',
  v4r: 'Yodan Verb',
  v5: 'Godan Verb',
  v5aru: 'Godan Verb',
  v5b: 'Godan Verb',
  v5g: 'Godan Verb',
  v5k: 'Godan Verb',
  'v5k-s': 'Godan Verb',
  v5m: 'Godan Verb',
  v5n: 'Godan Verb',
  v5r: 'Godan Verb',
  'v5r-i': 'Godan Verb',
  v5s: 'Godan Verb',
  v5t: 'Godan Verb',
  v5u: 'Godan Verb',
  'v5u-s': 'Godan Verb',
  v5uru: 'Godan Verb',
  vz: 'Ichidan Verb',
  vi: 'Intransitive Verb',
  vk: 'Kuru Verb',
  vn: 'Irregular Verb',
  vr: 'Irregular Verb',
  vs: 'Suru Verb',
  'vs-c': 'Su Verb',
  'vs-s': 'Suru Verb',
  'vs-i': 'Irregular Suru Verb',
  kyb: 'Kyoto-ben',
  osb: 'Osaka-ben',
  ksb: 'Kansai-ben',
  ktb: 'Kantou-ben',
  tsb: 'Tosa-ben',
  thb: 'Touhoku-ben',
  tsug: 'Tsugaru-ben',
  kyu: 'Kyuushuu-ben',
  rkb: 'Ryuukyuu-ben',
  nab: 'Nagano-ben',
  hob: 'Hokkaido-ben',
  vt: 'Transitive verb',
  vulg: 'Vulgar',
  'adj-kari': 'Adjective',
  'adj-ku': 'Adjective',
  'adj-shiku': 'Adjective',
  'adj-nari': 'Adjective',
  'n-pr': 'Proper Noun',
  'v-unspec': 'Verb',
  v4k: 'Yodan Verb',
  v4g: 'Yodan Verb',
  v4s: 'Yodan Verb',
  v4t: 'Yodan Verb',
  v4n: 'Yodan Verb',
  v4b: 'Yodan Verb',
  v4m: 'Yodan Verb',
  'v2k-k': 'Nidan Verb',
  'v2g-k': 'Nidan Verb',
  'v2t-k': 'Nidan Verb',
  'v2d-k': 'Nidan Verb',
  'v2h-k': 'Nidan Verb',
  'v2b-k': 'Nidan Verb',
  'v2m-k': 'Nidan Verb',
  'v2y-k': 'Nidan Verb',
  'v2r-k': 'Nidan Verb',
  'v2k-s': 'Nidan Verb',
  'v2g-s': 'Nidan Verb',
  'v2s-s': 'Nidan Verb',
  'v2z-s': 'Nidan Verb',
  'v2t-s': 'Nidan Verb',
  'v2d-s': 'Nidan Verb',
  'v2n-s': 'Nidan Verb',
  'v2h-s': 'Nidan Verb',
  'v2b-s': 'Nidan Verb',
  'v2m-s': 'Nidan Verb',
  'v2y-s': 'Nidan Verb',
  'v2r-s': 'Nidan Verb',
  'v2w-s': 'Nidan Verb',
  archit: 'Architecture',
  astron: 'Astronomy',
  baseb: 'Baseball',
  biol: 'Biology',
  bot: 'Botany',
  bus: 'Business',
  econ: 'Economics',
  engr: 'Engineering',
  finc: 'Finance',
  geol: 'Geology',
  law: 'Law',
  mahj: 'Mahjong',
  med: 'Medicine',
  music: 'Music',
  Shinto: 'Shinto',
  shogi: 'Shogi',
  sports: 'Sports',
  sumo: 'Sumo',
  zool: 'Zoology',
  joc: 'Joking',
  anat: 'Anatomical',
};
