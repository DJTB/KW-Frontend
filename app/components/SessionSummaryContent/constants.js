export const DEFAULT_ACTION = 'app/SessionSummaryPage/DEFAULT_ACTION';
