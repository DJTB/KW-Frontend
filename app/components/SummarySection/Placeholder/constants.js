export const PLACEHOLDERS = {
  CORRECT: {
    text: ' 満点！',
    kaomoji: ' (๑•̀ㅂ•́)و',
  },
  INCORRECT: {
    text: '零点',
    kaomoji: 'ʕノ•ᴥ•ʔノ ︵ ┻━┻',
  },
  CRITICAL: {
    text: '無事！',
    kaomoji: '(^・ω・^ )',
  },
};
