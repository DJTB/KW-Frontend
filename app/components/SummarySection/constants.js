export const CRITICAL = 'CRITICAL';
export const CORRECT = 'CORRECT';
export const INCORRECT = 'INCORRECT';

export const TYPES = {
  [CORRECT]: {
    color: 'green',
  },
  [INCORRECT]: {
    color: 'red',
  },
  [CRITICAL]: {
    color: 'orange',
  },
};
