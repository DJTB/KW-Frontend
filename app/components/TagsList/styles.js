import styled from 'styled-components';
import { resetList, fluidType } from 'shared/styles/utils';
import { halfElementGutterMobile, halfElementGutterDesktop, borderRadius } from 'shared/styles/sizing';
import { media } from 'shared/styles/media';
import * as COLORS from 'shared/styles/colors';

export const Ul = styled.ul`
  ${resetList}
  padding: ${halfElementGutterMobile};
  ${media('min').sm`
    padding: ${halfElementGutterDesktop};
  `}
`;

export const Li = styled.li`
  ${fluidType(10, 16, 300, 1800)}
  display: inline-flex;
  max-width: 100%;
  padding: .25rem .5rem .2rem;
  line-height: 1.2;
  text-decoration: none;
  vertical-align: middle;
  align-items: center;
  color: ${({ color }) => COLORS[color]};
  background-color: ${({ bgColor }) => COLORS[bgColor]};
  border-radius: ${borderRadius};
  margin: ${halfElementGutterMobile};
  ${media('min').sm`
    margin: ${halfElementGutterDesktop};
  `}
`;

export const Span = styled.span`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-left: .2em;
  margin-right: .2em;
`;
