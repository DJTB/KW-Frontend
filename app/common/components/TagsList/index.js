import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import cuid from 'cuid';

import { selectTags } from 'features/vocab/selectors';

import parseTags from 'common/utils/parseTags';
import getTagColors from './utils/getTagColors';

import { Ul, Li, Text } from './styles';

TagsList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string),
  isVisible: PropTypes.bool,
};

TagsList.defaultProps = {
  tags: [],
  isVisible: true,
};

export function TagsList({ tags, isVisible, ...props }) {
  const longformTags = parseTags(tags);
  return (
    <Ul isHidden={!isVisible} {...props}>
      {longformTags.map((text) => (
        <Li key={cuid()} {...getTagColors(text)}>
          <Text>{text}</Text>
        </Li>
      ))}
    </Ul>
  );
}

const mapStateToProps = (state, props) => ({
  tags: selectTags(state, props),
});

export default connect(mapStateToProps)(TagsList);
