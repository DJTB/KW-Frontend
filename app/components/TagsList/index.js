import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

import * as COLORS from 'shared/styles/colors';
import { PARTS_OF_SPEECH } from 'shared/constants';
import { Ul, Li, Span } from './styles';

const selectColors = ({ text, color, bgColor }) => {
  if (/common/i.test(text)) {
    return {
      color: 'whiteLight',
      bgColor: 'blue',
    };
  }
  if (/jlpt/i.test(text)) {
    return {
      color: 'blackLight',
      bgColor: 'tan',
    };
  }
  return {
    color,
    bgColor,
  };
};

TagsList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.oneOf(PARTS_OF_SPEECH)),
  color: PropTypes.oneOf(Object.keys(COLORS)),
  bgColor: PropTypes.oneOf(Object.keys(COLORS)),
};

TagsList.defaultProps = {
  tags: [],
  color: 'whiteLight',
  bgColor: 'grey',
};

function TagsList({ tags, ...props }) {
  return (
    <Ul {...props}>
      {tags.map((text) => (
        <Li key={cuid()} {...selectColors({ text, ...props })} >
          <Span>{text}</Span>
        </Li>
      ))}
    </Ul>
  );
}

export default TagsList;
