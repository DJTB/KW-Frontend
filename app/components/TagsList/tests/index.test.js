import React from 'react';
import { shallow } from 'enzyme';
import { PARTS_OF_SPEECH } from 'shared/constants';
import TagsList from '../index';

describe('<TagsList />', () => {
  it('should not render any children if no items provided', () => {
    expect(shallow(<TagsList />)).toMatchSnapshot();
  });
  it('should match PARTS_OF_SPEECH constants', () => {
    expect(shallow(<TagsList tags={PARTS_OF_SPEECH} />)).toMatchSnapshot();
  });
  it('should set colors for common and jlpt tags', () => {
    expect(shallow(<TagsList tags={['Common', 'JLPT N1']} />)).toMatchSnapshot();
  });
});
