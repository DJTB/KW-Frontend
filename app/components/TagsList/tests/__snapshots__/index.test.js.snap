// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsList /> should match PARTS_OF_SPEECH constants 1`] = `
<styled.ul
  bgColor="grey"
  color="whiteLight"
>
  <styled.li
    bgColor="tan"
    color="blackLight"
  >
    <styled.span>
      JLPT N1
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="tan"
    color="blackLight"
  >
    <styled.span>
      JLPT N2
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="tan"
    color="blackLight"
  >
    <styled.span>
      JLPT N3
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="tan"
    color="blackLight"
  >
    <styled.span>
      JLPT N4
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="tan"
    color="blackLight"
  >
    <styled.span>
      JLPT N5
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="blue"
    color="whiteLight"
  >
    <styled.span>
      Common
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Intransitive verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Noun
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Suru verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Transitive verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Ichidan verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      No adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Na adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Suffix
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      I adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Proper noun
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Pre-noun adjectival
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Adverb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Numeral
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Expression
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Prefix
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Pronoun
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Counter
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Interjection
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Conjunction
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      No-adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Na-adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Adverb taking the 'to' particle
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Noun - used as a suffix
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      intransitive verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      I-adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Kuru verb - special class
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Full name
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Suru verb - irregular
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Noun - used as a prefix
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Noun - used as a suffix
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Adverbial noun
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Temporal noun
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Place
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Auxiliary verb
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Noun or verb acting prenominally
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Taru-adjective
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb with ru ending
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb with u ending
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb with ku ending
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb with su ending
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb with mu ending
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="grey"
    color="whiteLight"
  >
    <styled.span>
      Godan verb with gu ending
    </styled.span>
  </styled.li>
</styled.ul>
`;

exports[`<TagsList /> should not render any children if no items provided 1`] = `
<styled.ul
  bgColor="grey"
  color="whiteLight"
/>
`;

exports[`<TagsList /> should set colors for common and jlpt tags 1`] = `
<styled.ul
  bgColor="grey"
  color="whiteLight"
>
  <styled.li
    bgColor="blue"
    color="whiteLight"
  >
    <styled.span>
      Common
    </styled.span>
  </styled.li>
  <styled.li
    bgColor="tan"
    color="blackLight"
  >
    <styled.span>
      JLPT N1
    </styled.span>
  </styled.li>
</styled.ul>
`;
